src/app/extensions/quickorder/shared/direct-order/direct-order.component.ts

```diff
@@ -0,0 +1,122 @@
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable, Subject } from 'rxjs';
+import { debounceTime, mapTo, switchMap, takeUntil, tap } from 'rxjs/operators';
+
+import { ProductContextFacade } from 'ish-core/facades/product-context.facade';
+import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
+import { ProductCompletenessLevel, ProductHelper } from 'ish-core/models/product/product.helper';
+import { GenerateLazyComponent } from 'ish-core/utils/module-loader/generate-lazy-component.decorator';
+
+@Component({
+  selector: 'ish-direct-order',
+  templateUrl: './direct-order.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+@GenerateLazyComponent()
+export class DirectOrderComponent implements OnInit, OnDestroy {
+  model = { sku: '', quantity: 1 };
+  directOrderForm = new FormGroup({
+    sku: new FormControl(this.model.sku),
+    quantity: new FormControl(this.model.quantity),
+  });
+  disabledAddToCart = true;
+  fields: FormlyFieldConfig[] = [
+    {
+      key: 'sku',
+      type: 'ish-text-input-field',
+      className: 'col-12 list-item search-container mb-1',
+      templateOptions: {
+        placeholder: 'shopping_cart.direct_order.item_placeholder',
+        autocomplete: true,
+      },
+      expressionProperties: {
+        'templateOptions.required': () => (!this.directOrderForm.untouched ? !!this.model.quantity : false),
```

Comment:
read the quantity and the quantityError from the context facade, see product-add-to-basket component this.hasQuantityError$ = this.context.select('hasQuantityError');